keras_cv/layers/preprocessing/batched_base_image_augmentation_layer.py

```diff
@@ -0,0 +1,313 @@
+        images = inputs.get(IMAGES, None)
+        labels = inputs.get(LABELS, None)
+        bounding_boxes = inputs.get(BOUNDING_BOXES, None)
+        keypoints = inputs.get(KEYPOINTS, None)
+        segmentation_masks = inputs.get(SEGMENTATION_MASKS, None)
+
+        batch_size = tf.shape(images)[0]
+
+        transformations = self.get_random_transformation_batch(
+            batch_size,
+            images=images,
+            labels=labels,
+            bounding_boxes=bounding_boxes,
+            keypoints=keypoints,
+            segmentation_masks=segmentation_masks,
+        )
+
+        images = self.augment_images(
+            images,
+            transformations=transformations,
+            bounding_boxes=bounding_boxes,
+            label=labels,
+        )
+
+        result = {IMAGES: images}
+        if labels is not None:
+            labels = self.augment_targets(
+                labels,
+                transformations=transformations,
+                bounding_boxes=bounding_boxes,
+                image=images,
+            )
+            result[LABELS] = labels
+
+        if bounding_boxes is not None:
+            bounding_boxes = self.augment_bounding_boxes(
+                bounding_boxes,
+                transformations=transformations,
+                labels=labels,
+                images=images,
+            )
+            bounding_boxes = bounding_box.to_ragged(bounding_boxes)
+            result[BOUNDING_BOXES] = bounding_boxes
+
+        if keypoints is not None:
+            keypoints = self.augment_keypoints(
+                keypoints,
+                transformations=transformations,
+                label=labels,
+                bounding_boxes=bounding_boxes,
+                images=images,
+            )
+            result[KEYPOINTS] = keypoints
+        if segmentation_masks is not None:
+            segmentation_masks = self.augment_segmentation_masks(
+                segmentation_masks,
+                transformations=transformations,
+            )
+            result[SEGMENTATION_MASKS] = segmentation_masks
+
+        # preserve any additional inputs unmodified by this layer.
+        for key in inputs.keys() - result.keys():
+            result[key] = inputs[key]
+        return result
+
+    def call(self, inputs, training=True):
+        inputs = self._ensure_inputs_are_compute_dtype(inputs)
+        if training:
+            inputs, metadata = self._format_inputs(inputs)
+            images = inputs[IMAGES]
+            if images.shape.rank == 3 or images.shape.rank == 4:
+                return self._format_output(self._batch_augment(inputs), metadata)
+            else:
+                raise ValueError(
+                    "Image augmentation layers are expecting inputs to be "
+                    "rank 3 (HWC) or 4D (NHWC) tensors. Got shape: "
+                    f"{images.shape}"
+                )
+        else:
+            return inputs
+
+    def _format_inputs(self, inputs):
```

Comment:
there are so many other layers on the base layer, and this will eventually replace the base layer if we finish migrating every layer.  

I don't think the case here is strong enough for this to be a subclass.

Reply:
actually this has some different behavior in it too - in the expand_dims() section below.